### PR TITLE
Switch Sphinx theme.

### PR DESCRIPTION
**Pull request by gassmoeller**
Remade version of #373 to trigger the readthedocs build for this PR. I will need to play around with some options, so I will keep this as work in progress for now.

### CODE REVIEW

**Comment by gassmoeller** [comment]
Ok, it looks like I fixed all the bits and pieces that came up. Additional fixes I needed:
- Fix some versions in the docker container
- Fix a github action issue
- Switch the Dockerfile back to OpenMPI (from MPICH). The container on docker hub actually always used OpenMPI, I do not know why the Dockerfile was set to MPICH. I will work on update the containers anyway later this week so this shouldnt be an issue.

TLDR: Check if this looks good, and if it does this is ready to merge. New documentation layout:
https://rayleigh-documentation--378.org.readthedocs.build/en/378/

**Review by feathern — APPROVED** [review]
Looks great!
